src/components/BeerDetails.js

```diff
@@ -0,0 +1,46 @@
+import  axios  from 'axios';
+import { useEffect, useState } from 'react';
+import { useParams  } from 'react-router-dom';
+
+function BeerDetails() {
+  
+    const {id} = useParams();
+
+    const [details, setDetails] = useState(null)
+
+    const baseUrl = 'https://ih-beers-api2.herokuapp.com/beers';
+       
+    useEffect(() => {
+      axios
+        .get(baseUrl + id)
+        .then((response) => {
+          setDetails(response.data);
+        })
+        .catch((err) => {
+          console.log(err);
+        });
+    }, []);
+  
+  
+
+  return (
+    <>
+      {beers.map((beer) => {
+          console.log(response)
+        return (
+            <>
+            <img src={details.image_url}/>
+            <h2>{details.name}</h2>
+            <h3>{details.tagline}</h3>
+            <h3>{details.first_brewed}</h3>
+            <h3>{details.attenuation_level}</h3>
+            <h3>{details.description}</h3>
+            <h3>{details.contributed_by}</h3>         
+          </>
+        );
+      })}
```

Comment:
- This is beerDetails component, where we fetch details of one beer from database, so there is not array to iterate over , in this case you don't even have variable `beers` in this file.

- One more point here, remember that in react it takes a moment to fetch the data, so you only want to render information if you already have it! You must use conditional rendering, otherwise you'll get an error in terminal. For example something like this:

```
{details &&
            <>
            <img src={details.image_url}/>
            <h2>{details.name}</h2>
            <h3>{details.tagline}</h3>
            <h3>{details.first_brewed}</h3>
            <h3>{details.attenuation_level}</h3>
            <h3>{details.description}</h3>
            <h3>{details.contributed_by}</h3>         
          </>
        }
```